Subject: Make piper TTS optional when onnxruntime is not available

onnxruntime is not packaged for Solus, so make the piper extension
optional by adding an error check similar to podofo.

--- a/setup/build_environment.py	2026-01-30 08:56:37.748463095 -0500
+++ b/setup/build_environment.py	2026-01-30 08:56:37.751970114 -0500
@@ -249,5 +249,8 @@
         ('PoDoFo not found on your system. Various PDF related',
     ' functionality will not work. Use the PODOFO_INC_DIR and',
     ' PODOFO_LIB_DIR environment variables.')
+piper_error = None if any(os.path.exists(os.path.join(d, 'onnxruntime_cxx_api.h')) for d in piper_inc_dirs) else \
+        ('onnxruntime not found on your system. Piper TTS functionality will not be available.',)
+
 podofo_inc_dirs = [podofo_inc, os.path.dirname(podofo_inc)]
 podofo_lib_dirs = [podofo_lib]
--- a/setup/extensions.json	2026-01-30 08:56:50.886520661 -0500
+++ b/setup/extensions.json	2026-01-30 09:00:50.818987768 -0500
@@ -138,6 +138,8 @@
         "name": "piper",
         "sources": "calibre/utils/tts/piper.cpp",
 		"needs_c++": "17",
+        "error": "!piper_error",
+        "optional": true,
         "libraries": "!piper_libs",
         "lib_dirs": "!piper_lib_dirs",
         "inc_dirs": "!piper_inc_dirs"
